docs/proposals/20201125-jaeger-traces-store.md

```diff
@@ -0,0 +1,122 @@
+
+## Why
+
+Traces store adds a crucial capability to Observatorium to persist and query tracing data.
+
+## Goals
+
+* Configure OpenTelemetry collector to export data to Jaeger collector.
+* Deploy Jaeger with a persistent storage.
+* Define query API for trace data.
+
+## Non-Goals
+
+* Define trace ingestion API - it is done in [Traces ingestion API and OpenTelemetry collector](https://github.com/observatorium/observatorium/pull/443) proposal.
+
+## How
+
+The how is split into the following sections:
+* Jaeger configuration - describes how Jaeger components are deployed.
+* Jaeger Query - describes how users access data stored in Jaeger.
+
+### Jaeger configuration
+
+Currently, the most adopted and supported storage in Jaeger is Elasticsearch. Red Hat also supports Jaeger with Elasticsearch (6.x) as part of the OpenShift product. Therefore, this proposal assumes Elasticsearch will be used as a storage backend. Distributed tracing team at Red Hat is looking at alternative storages hence the storage technology can change in the future.
+
+Follows possible multitenant Jaeger deployment topologies. All deployment topologies assume that Jaeger will be deployed by the [Jaeger Operator](https://github.com/jaegertracing/jaeger-operator) which depends on [OpenShift Elasticserach operator](https://github.com/openshift/elasticsearch-operator) and [Strimzi operator](https://operatorhub.io/operator/strimzi-kafka-operator) if Kafka is used. Jaeger can be as well configured to use externally managed Elasticsearch or Kafka instances.
+
+#### Single Jaeger instance
+
+This deployment topology deploys a single Jaeger instance for all tenants with a single Elasticsearch instance. The data for each tenant would contain a unique attribute to identify a tenant. The label would be dynamically injected in the Observatorium API service or in the OpenTelemetry collector.
+
+This deployment strategy will not work with the current Jaeger components. Here are few issues, and the list might not be complete:
+* Service and operation name API. The service query API nor storage does not expose/store labels.
+* Dependency/service architecture diagram. To support this feature dependency schema, query, collector and Spark aggregation would have to change.
+* Data retention/TTL configurable pre tenant. The data retention in Elasticsearch is configurable per index.
+
+##### OpenTelemetry collector configuration
+
+The OpenTelemetry collector is configured with a single exporter to export data to a single Jaeger collector.
+
+#### Jaeger instance per tenant
+
+This deployment topology deploys a Jaeger instance per tenant and all Jaeger instances use a single Elasticsearch instance. The data for each tenant is stored in a separate Elasticsearch index. Jaeger was designed to separate tenant data per index. Therefore, all Jaeger functionality works well with this deployment strategy.
+
+##### OpenTelemetry collector configuration
```

Comment:
That's not really a soft-tenancy, but could work as a  workaround for now. I think such simple implementation could be embedded directly to Obs API too